sdv/multi_table/base.py

```diff
@@ -28,15 +28,18 @@ class BaseMultiTableSynthesizer:
 
     _synthesizer = GaussianCopulaSynthesizer
     _numpy_seed = 73251
+    _numpy_generator = None
 
     @contextlib.contextmanager
     def _set_temp_numpy_seed(self):
         initial_state = np.random.get_state()
         if isinstance(self._numpy_seed, int):
             np.random.seed(self._numpy_seed)
+            self._numpy_generator = np.random.PCG64(self._numpy_seed)
```

Comment:
What exactly is going on here with the numpy generator?

Reply:
I found that at some point `scipy.stats` uses `numpy.random.default_rng` to control randomness
rather than the seeded one which is working widely across our projects.

Here is how `numpy` proposes the new randomness: https://numpy.org/doc/stable/reference/random/generator.html
which I think that down the line somewhere `scipy.stats` started to use it rather than using
the `numpy.random`.

The problem with the `default_rng` is that I didn't find a way to properly set a `numpy.random.get_state`
object, which ended up leading me to this solution.

Also I found this interesting, but it will require us to re-work part of `RDT`: [Good RNG practices](https://albertcthomas.github.io/good-practices-random-number-generators/),
TLDR; pass down the line a `generator` object rather than setting a `random.seed` and use `generator.random`
to generate random values etc...